bigframes/core/joins/merge.py

```diff
@@ -18,15 +18,18 @@
 
 from __future__ import annotations
 
+import typing
 from typing import Literal, Optional
 
-from bigframes.dataframe import DataFrame
-from bigframes.series import Series
+# Avoid cirular imports.
+if typing.TYPE_CHECKING:
+    import bigframes.dataframe
```

Comment:
I like disambiguating "DataFrame..." should we just always qualify it?

Reply:
Yes we should. I have a bug open to do that. Issue 296390934. Fully qualified names will also help with docs rendering.